pyproject.toml

```diff
@@ -31,15 +31,18 @@ classifiers = [
 python = ">=3.9,<4"
 pytest-cov = "^4.0.0"
 openmock = "^2.3.0"
+requests-mock = "^1.11.0"
+freezegun = "^1.2.2"
 
 # Optional dependencies - install with `poetry install -E docs`
 sphinx = {version="^7.1.0", optional=true}
 myst-parser = {version="^2.0.0", optional=true}
 pydata-sphinx-theme = {version="^0.13.3", optional=true}
+requests-aws4auth = "^1.2.3"
```

Comment:
This shouldn't be necessary since `requests-aws4auth` is in the lamda-dev dependency group

Reply:
Are you saying we don't need this because lambda is running inside AWS environment?

Reply:
The dependency is created again a few lines down, so it's a duplicate. One or the other should probably be removed.

Reply:
I do use this library in the unit test. If I keep it in pyproject.toml and remove it from requirements.txt, it will still be available in the lambdas?

Reply:
Yes, the requirements.txt export statement in `.pre-commit-config.yaml` does export the `lambda-dev` dependency group. Since the dependency is in that group, it is still available for install. I'm not sure what groups the unit tests use, but it should be easy to install the lambda-dev group for the tests as well. 

Or, if you just want to make this a greater project dependency, which would also be fine, it should be moved up out of the optional dependencies section, and the duplicate should be removed from `lambda-dev`. 

It's ok to have duplicate dependencies in different dependency groups. But, if it's a full-project dependency (under `[tool.poetry.dependencies]` then it is unnecessarily duplicated by putting it in a group. The full project dependencies are installed every time, while the groups can be included or excluded depending on the use case.

Reply:
I think I'm doing something wrong here/not understanding. I have the requests-as4auth under the lambda section in the pyproject.toml:
```
[tool.poetry.group.lambda-dev.dependencies]
opensearch-py = "^2.1.1"
requests = "^2.28.2"
requests-aws4auth = "^1.2.3"
```
but when I deploy and run the lambda, it's saying that requests-aws4auth can't be found when it tries to import. It looks like the requests and opensearch-py are in the requirements.txt file. Is requests-aws4auth supposed to be automatically added to that file at some point?

Reply:
The whole lambda-dev dependency group is automatically added to the requirements.txt file on pre-commit, which you can see [here](https://github.com/IMAP-Science-Operations-Center/sds-data-manager/blob/dev/.pre-commit-config.yaml#L32C26-L32C26). It could be that you need to manually run that export before you deploy - which you could do with the command `poetry export -f requirements.txt -o sds_data_manager/lambda_code/requirements.txt --only lambda-dev`. Or, by committing your change prior to deploying and triggering the pre-commit checks.

Reply:
Ah, fixed it! Thank you!